src/mesh/impls/bout/boutmesh.cxx

```diff
@@ -1868,20 +1868,22 @@ int BoutMesh::pack_data(const vector<FieldData *> &var_list, int xge, int xlt, i
   for (const auto &var : var_list) {
     if (var->is3D()) {
       // 3D variable
-      ASSERT2(static_cast<Field3D *>(var)->isAllocated());
-      for (int jx = xge; jx != xlt; jx++) {
-        for (int jy = yge; jy < ylt; jy++) {
-          for (int jz = 0; jz < LocalNz; jz++, len++) {
-            buffer[len] = (*static_cast<Field3D *>(var))(jx, jy, jz);
+      ASSERT2(var->isAllocated());
+      Indices j;
+      for (j.x = xge; j.x != xlt; j.x++) {
+        for (j.y = yge; j.y < ylt; j.y++) {
+          for (j.z = 0; j.z < LocalNz; j.z++, len++) {
+            buffer[len] = var->operator[](j);
```

Comment:
Why `var->operator[](j)` here and `(*var)[j]` below?

Reply:
No real reason. Do you have any preference?